File: stack/590. N 叉树的后序遍历.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[]}
 */
 var postorder = function(root) {
    let ans = [];
    post(root ,ans);
    return ans;
};

function post(root,ans){
    if(root == null) return ;
    if(root.children !== null){
        for(let i = 0; i < root.children.length; i++){
                post(root.children[i],ans)
            }
    }
    
    ans.push(root.val);
    return ;
}

File: Python/590. N 叉树的后序遍历.py
# -*- codeing = utf-8 -*-
# @Time : 2023-8-28 20:05
# @Author : 刘永奇
# @File : 590. N 叉树的后序遍历.py
# @Software : PyCharm
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        res = []
        self.post(root,res)
        return res

    def post(self,root,res):
        if not root:
            return
        for item in root.children:
            self.post(item,res)
        res.append(root.val)